### PR TITLE
Support profiling for keda components

### PR DESCRIPTION
**Pull request by yuvalweber**
_Provide a description of what has been changed_

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #

Relates to #4789

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -60,8 +60,8 @@ Here is an overview of all new **experimental** features:
 
 ### Improvements
 
-- **General**: Support profiling for keda components ([#4789](https://github.com/kedacore/keda/issues/4789))
 - **General**: Add parameter queryParameters to prometheus-scaler ([#4962](https://github.com/kedacore/keda/issues/4962))
+- **General**: Support profiling for keda components ([#4789](https://github.com/kedacore/keda/issues/4789))
```
Comment (zroubalik) [inline CHANGELOG.md:64]:
```suggestion
- **General**: Support profiling for KEDA components ([#4789](https://github.com/kedacore/keda/issues/4789))
```

**Comment by yuvalweber** [comment]
Hey @zroubalik can you please inform me what is missing in all of the PRs I opened (kedacore/charts#549, kedacore/keda-docs#1250, #5091 ) in order to merge everything?

**Comment by tomkerkhove** [comment]
> Hey @zroubalik can you please inform me what is missing in all of the PRs I opened ([kedacore/charts#549](https://github.com/kedacore/charts/pull/549), [kedacore/keda-docs#1250](https://github.com/kedacore/keda-docs/pull/1250), #5091 ) in order to merge everything?

@yuvalweber The e2e tests are failing, can you please check?

> /run-e2e internal **Update:** You can check the progress [here](https://github.com/kedacore/keda/actions/runs/6560538679)

**Comment by yuvalweber** [comment]
Hey @tomkerkhove can you help me a bit?
I looked at the errors and I see things failing because it's trying to create resources on namespaces that being terminated or something with rate limit, can you please help me figure out if it's related to my change (cause it shouldn't)?

**Comment by yuvalweber** [comment]
Hey @tomkerkhove would love for also update on this

**Comment by zroubalik** [comment]
> Hey @tomkerkhove can you help me a bit? I looked at the errors and I see things failing because it's trying to create resources on namespaces that being terminated or something with rate limit, can you please help me figure out if it's related to my change (cause it shouldn't)?

Env problem, the re-run passed 🚀

**Review by zroubalik — APPROVED** [review]

**Comment by yuvalweber** [comment]
Hey @zroubalik do I need to write test?
Sorry i'm just not familiar with writing tests in golang and not sure if necessary in this situation

**Comment by zroubalik** [comment]
> Hey @zroubalik do I need to write test? Sorry i'm just not familiar with writing tests in golang and not sure if necessary in this situation

@yuvalweber I don't think tests are needed for this kind of feature. On the other hand a short blogpost on how you used this profiler would be awesome 😃

**Comment by yuvalweber** [comment]
Would love to do it 😄 .
Can you share example of blogpost so I could see the format and know how to copy it?
I could share the example related to the issue I had before #4687

**Comment by zroubalik** [comment]
> Would love to do it 😄 . Can you share example of blogpost so I could see the format and know how to copy it? I could share the example related to the issue I had before #4687

DM me on slack :)

**Comment by yuvalweber** [comment]
Hey @zroubalik need you to rerun the e2e tests.
I wanted to merge PR and rebase but accidentally just updated by branch which triggered this new requirement for this test

**Comment by yuvalweber** [comment]
Hey @zroubalik can you just help me understand why it won't let me merge this?

**Comment by zroubalik** [comment]
> Hey @zroubalik can you just help me understand why it won't let me merge this?

Only maitiainers can merge the PR, we usually do that once all PRs (feature/charts/docs) are ready.

**Comment by yuvalweber** [comment]
Alright so when @tomkerkhove will finish reviewing my changes I'll be happy to merge.
Meanwhile I'll work on the blogpost

**Comment by yuvalweber** [comment]
> > Alright so when @tomkerkhove will finish reviewing my changes I'll be happy to merge. Meanwhile I'll work on the blogpost
> 
> You don't have to do anything on the PRs (only additional fixes requested by reviews). We will handle the rest

I think I addressed to all of the requests you guys reviewed but if I missed something please let me know 😄

**Pull request merged**